env/io_posix.cc

```diff
@@ -409,13 +370,135 @@ size_t PosixHelper::GetUniqueIdFromFile(int fd, char* id, size_t max_size) {
   return static_cast<size_t>(rid - id);
 }
 #endif
+
+#ifdef OS_LINUX
+void LogicalBufferSizeCache::RefAndCacheLogicalBufferSize(
+    const std::set<std::pair<std::string, int>>& directories) {
+  std::map<int, size_t> fd_sizes;
+  {
+    ReadLock lock(&cache_mutex_);
+    for (auto& fname_fd : directories) {
+      if (cache_.find(fname_fd.first) == cache_.end()) {
+        fd_sizes.emplace(fname_fd.second, 0);
```

Comment:
I chose `emplace` because I think `fb_sizes[fname_fd.second]` alone already constructs and inserts the `{fname_fd.second, 0}` pair, then ` = 0` is an extra copy, I want to avoid that extra copy.

Reply:
I am not expert on it but I feel that assigning an integer has a very minor cost. Maybe ask someone who is more familiar with it. But it's your call here.